### PR TITLE
Update link / base velocities and accelerations

### PR DESCRIPTION
**Pull request by diegoferigo**
- Fix a copy-and-paste bug ea0d34f 
- Improve the base link APIs clarifying the velocity representation e7b9b1e 
- Get the body base velocity e7b9b1e 
- Add new test that check the link and base APIs for velocity and accelerations against numerically differentiated values 830b30a

Referring to [Multibody Dynamics Notation](https://pure.tue.nl/ws/portalfiles/portal/139293126/A_Multibody_Dynamics_Notation_Revision_2_.pdf), the `World` methods use the _mixed velocity_ representation [Section 5.2] and the `Body` methods the _left trivialized_ representation (body fixed frame) [Section 5].

Related to https://github.com/ignitionrobotics/ign-gazebo/issues/87.

### CODE REVIEW

File: tests/test_scenario/test_link_velocities.py [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+    for _ in range(int(0.5 / dt)):
+
+        position_old = np.array(position())
+        assert gazebo.run()
+        position_new = np.array(position())
+
+        world_velocity = (position_new - position_old) / dt
+
+        # Test the world velocity (MIXED)
+        assert world_velocity == pytest.approx(worldLinearVelocity(), abs=1E-2)
```
Comment (traversaro) [inline tests/test_scenario/test_link_velocities.py:127]: `1E-2` is quite high, what are the velocity values ?

File: tests/test_scenario/test_link_velocities.py [hunk 2]
```diff
@@ -0,0 +1,314 @@
...
+
+        velocity_old = np.array(worldLinearVelocity())
+        assert gazebo.run()
+        velocity_new = np.array(worldLinearVelocity())
+
+        world_acceleration = (velocity_new - velocity_old) / dt
+
+        # By time to time there are steps where the acceleration becomes extremely high,
+        # like 1000 m/s/s when the average is never exceeds 4 m/s/s.
+        # We exclude those points. We should understand why this happens.
```
Comment (traversaro) [inline tests/test_scenario/test_link_velocities.py:241]: 😨

File: tests/test_scenario/test_link_velocities.py (hunk 1 above)
Comment (diegoferigo) [reply tests/test_scenario/test_link_velocities.py:127]: The z component of this velocity reaches ~4, x and y are much smaller, less than 1.0.

File: tests/test_scenario/test_link_velocities.py (hunk 2 above)
Comment (diegoferigo) [reply tests/test_scenario/test_link_velocities.py:241]: Not sure what's happening there :thinking:

Comment (diegoferigo) [reply tests/test_scenario/test_link_velocities.py:241]: There are only few cases in `0.5 sec * 10kH = 5k runs`

**Pull request merged**